onnxruntime/python/tools/transformers/shape_infer_helper.py

```diff
@@ -0,0 +1,115 @@
+import os
+import sys
+
+# In ORT Package the symbolic_shape_infer.py is in ../tools
+file_path = os.path.dirname(__file__)
+if os.path.exists(os.path.join(file_path, "../tools/symbolic_shape_infer.py")):
+    sys.path.append(os.path.join(file_path, '../tools'))
+else:
+    sys.path.append(os.path.join(file_path, '..'))
+from symbolic_shape_infer import *
+
+import onnx
+
+
+class SymbolicShapeInferenceHelper(SymbolicShapeInference):
+    def __init__(self, model, verbose=0, int_max=2**31 - 1, auto_merge=True, guess_output_rank=False):
+        super().__init__(int_max, auto_merge, guess_output_rank, verbose)
+        self.model_ = onnx.ModelProto()
+        self.model_.CopyFrom(model)
+
+        self.dispatcher_['Gelu'] = self._infer_Gelu
+        self.dispatcher_['FastGelu'] = self._infer_FastGelu
+        self.dispatcher_['BiasGelu'] = self._infer_BiasGelu
+        self.dispatcher_['Attention'] = self._infer_Attention
+        self.dispatcher_['LongformerAttention'] = self._infer_LongformerAttention
+        self.dispatcher_['LayerNormalization'] = self._infer_LayerNormalization
+        self.dispatcher_['SkipLayerNormalization'] = self._infer_SkipLayerNormalization
+
+        self.all_shapes_inferred_ = False
+        self.inferred_ = False
+
+    # The goal is to remove dynamic_axis_mapping
+    def infer(self, dynamic_axis_mapping={"batch_size": 4, "seq_len": 7}):
```

Comment:
>dynamic_axis_mapping={"batch_size": 4, "seq_len": 7} [](start = 20, length = 52)

Seems the intention is to replace dynamic axis with integer values. Usually I edit the graph input to replace dims directly, and then run symbolic shape inference, like in onnxruntime-nuphar-tutorial section 5.2:
```
# edit batch dimension from symbolic to int value for better codegen
mp = onnx.load(gpt2_model)
mp.graph.input[0].type.tensor_type.shape.dim[0].dim_value = 1
onnx.save(mp, gpt2_model)
```

Reply:
yeah but this would be more model specific